### PR TITLE
feat(share, config): more variables, placeholder and reset default

### PR DESCRIPTION
**Pull request by iUnstable0**
Feature request #94 

Added more variables for the share email option
- {desc} - Share description
- {expires} - Share expiration time (ex: in 6 hours)

Config input now shows default value in placeholder
To reset to default just save an empty string

https://user-images.githubusercontent.com/46888825/226200101-3f6b7084-9474-438b-a3cf-a1489e45acdb.mp4

Also edited some default value in prisma seed file

### CODE REVIEW

**Comment by stonith404** [comment]
I've just removed the `luxon` package as we already use `moment` in the project.

Additionally I also removed the text "... so make sure you download your files before then." because it would sound really confusing if someone set the expiration to "never":
image

**Comment by stonith404** [comment]
I'm not sure if we should rename `editedValue` to `value` and `value` to `defaultValue` to make it a bit clearer. What do you think?

**Comment by iUnstable0** [comment]
yeah sure, also one thing in the moment : "never" it would be
This share will expire never
maybe change it to like

```
" in: never"
```
?

**Comment by stonith404** [comment]
> yeah sure, also one thing in the moment : "never" it would be This share will expire never maybe change it to like
> 
> ```
> " in: never"
> ```
> 
> i'm assuming your moment date thing will return something like `in xx days`
> 
> it sounds better 😄

Sorry, I'm not sure if I understand what you mean. So you would replace `This share will expire never` with `This share will expire in: never`?

**Comment by Hukuma1** [comment]
Hope it's okay to jump in with some more info. Since this is for e-mails, it might be beneficial to show the actual expiration date/time? Since it's static, not dynamic.

Otherwise if you open the e-mail 6 hours from when it was sent, the expiration will be 6 hours behind. And the person opening the e-mail would have to do the math based on when they received it? Unless I'm misunderstanding as well. ;)

**Comment by iUnstable0** [comment]
it wouldn't be possible because of time zones

**Comment by Hukuma1** [comment]
Makes sense. Perhaps let’s default to server time zone and write it in? E.g. Expires: March 21st 11:00am PST. If someone accesses the server share maybe it can be dynamically updated using their local time (if possible), but for email this should suffice I think.

**Comment by iUnstable0** [comment]
> > yeah sure, also one thing in the moment : "never" it would be This share will expire never maybe change it to like
> > ```
> > " in: never"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > i'm assuming your moment date thing will return something like `in xx days`
> > it sounds better 😄
> 
> Sorry, I'm not sure if I understand what you mean. So you would replace `This share will expire never` with `This share will expire in: never`?

yes, so the template would be

This share will expire {expires}
If the date is not set then return ": never" as it sounds more correct

**Comment by iUnstable0** [comment]
> Yeah, makes sense. I only share with same time zone typically, so it doesn't affect me the same way as others might be. Thanks for working on this!

oh then maybe I could make an option like a toggle for that?

**Comment by Hukuma1** [comment]
I don't want to over-complicate it. Let's just go with your idea first and if others ask for it later, maybe? :D

**Comment by iUnstable0** [comment]
since you changed value to default null you will have to change the code if value === "" to like if !value

File: frontend/src/pages/admin/config/[category].tsx [hunk 1]
```diff
@@ -67,16 +67,20 @@ export default function AppShellDemo() {
           toast.success("Configurations updated successfully");
         })
         .catch(toast.axiosError);
-      config.refresh();
+      void config.refresh();
```
[the remark this answers is not in the document]
Comment (iUnstable0) [reply frontend/src/pages/admin/config/[category].tsx:70]: oh I just added it to tell my IDE that I'm intentionally ignoring await on async function

[the remark this answers is not in the document]
Comment (stonith404) [reply frontend/src/pages/admin/config/[category].tsx:70]: Ah okay 👍

**Review by stonith404 — APPROVED** [review]

**Comment by stonith404** [comment]
@iUnstable0 I'm not really sure, do you have the rights to merge the PR, or do I have to do it?

**Pull request merged**